homeassistant/components/ezviz/alarm_control_panel.py

```diff
@@ -0,0 +1,166 @@
+        DATA_COORDINATOR
+    ]
+
+    async_add_entities([EzvizAlarm(coordinator, entry.entry_id, ALARM_TYPE)])
+
+
+class EzvizAlarm(CoordinatorEntity, AlarmControlPanelEntity):
+    """Representation of an Ezviz alarm control panel."""
+
+    coordinator: EzvizDataUpdateCoordinator
+    entity_description: EzvizAlarmControlPanelEntityDescription
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_supported_features = (
+        AlarmControlPanelEntityFeature.ARM_AWAY
+        | AlarmControlPanelEntityFeature.ARM_HOME
+    )
+    _attr_code_arm_required = False
+
+    def __init__(
+        self,
+        coordinator: EzvizDataUpdateCoordinator,
+        entry_id: str,
+        entity_description: EzvizAlarmControlPanelEntityDescription,
+    ) -> None:
+        """Initialize alarm control panel entity."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{entry_id}_{entity_description.key}"
+        self.entity_description = entity_description
+        self._attr_device_info: DeviceInfo = {
+            "identifiers": {(DOMAIN, "EZVIZ Alarm")},
+            "name": "EZVIZ Alarm",
+            "model": "EZVIZ Alarm",
+            "manufacturer": MANUFACTURER,
+        }
+        self._attr_state = None
+
+    async def async_added_to_hass(self) -> None:
+        """Entity added to hass."""
+        self.hass.async_add_executor_job(self.update)
+
+    def alarm_disarm(self, code: str | None = None) -> None:
+        """Send disarm command."""
+        try:
+            if self.coordinator.ezviz_client.api_set_defence_mode(
+                DefenseModeType.HOME_MODE.value
+            ):
+                self._attr_state = STATE_ALARM_DISARMED
+                self.async_write_ha_state()
+
+        except PyEzvizError as err:
+            raise HomeAssistantError("Cannot disarm EZVIZ alarm") from err
+
+    def alarm_arm_away(self, code: str | None = None) -> None:
+        """Send arm away command."""
+        try:
+            if self.coordinator.ezviz_client.api_set_defence_mode(
+                DefenseModeType.AWAY_MODE.value
+            ):
+                self._attr_state = STATE_ALARM_ARMED_AWAY
+                self.async_write_ha_state()
+
+        except PyEzvizError as err:
+            raise HomeAssistantError("Cannot arm EZVIZ alarm") from err
+
+    def alarm_arm_home(self, code: str | None = None) -> None:
+        """Send arm home command."""
+        try:
+            if self.coordinator.ezviz_client.api_set_defence_mode(
+                DefenseModeType.SLEEP_MODE.value
+            ):
+                self._attr_state = STATE_ALARM_ARMED_HOME
+                self.async_write_ha_state()
+
+        except PyEzvizError as err:
+            raise HomeAssistantError("Cannot arm EZVIZ alarm") from err
+
+    def update(self) -> None:
+        """Fetch data from EZVIZ."""
+        _LOGGER.debug("Updating %s", self.name)
+        ezviz_alarm_state_number: str = "0"
```

Comment:
typing is superflues.
```suggestion
        ezviz_alarm_state_number = "0"
```